src/main/java/me/zeroeightsix/kami/gui/rgui/component/ToggleButton.kt

```diff
@@ -0,0 +1,27 @@
+package me.zeroeightsix.kami.gui.rgui.component
+
+import me.zeroeightsix.kami.util.math.Vec2f
+
+class ToggleButton(
+    name: String,
+    description: String,
+    val check: () -> Boolean,
+    val toggle: () -> Unit)
+    : BooleanSlider(name, 0.0, description, { true }) {
+
+    init {
+        if (check()) value = 1.0
+    }
+
+    override fun onTick() {
+        super.onTick()
+        value = if (check()) 1.0 else 0.0
+    }
```

Comment:
I'm not sure if this is the best solution, it might be better to add a field to BooleanSliders if they're changable or not? :thinking: